### DISCUSSION TITLE
Allow `python` like `*args **kwargs`.

### DISCUSSION THREAD

**Discussion by bukowa** [casey/just#2197] [Ideas]
I think it would be a good idea to use something like that:

```justfile
go *args, **kwargs:
    echo {{args[0]}}
    echo {{kwargs['ok']}}
````

**Comment by runeimp** [comment]
`*args` is essentially available via the `positional-arguments` setting which will make `$@`, `$0`, `$1`, etc. available to most non-Windows shells. See [Settings - Just Programmer's Manual](https://just.systems/man/en/chapter_27.html#positional-arguments)

`**kwargs` exists in Python because Python has always allowed for named arguments, which I absolutely love. Just does not. It does already allows setting of environment variables in a `kwargs` way though. See [Setting Variables from the Command Line](https://just.systems/man/en/chapter_39.html#setting-variables-from-the-command-line). Changing this behavior to named arguments instead would be a breaking change. I don't know how often that feature is used but Just has been around for several years now and it is used in a lot of CI/CD systems. I'd rather have that feature be for named arguments instead of named environment variables but that ship sailed a long time ago.

Note: the chapter numbers referenced in the links to the manual could change in the future

**Reply by bukowa** [reply]
I understand, I just think Python did it the best, simplest and prettiest way, and it couldn't be better ^^
